src/mococrw/mac.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+#include <memory>
+#include <vector>
+#include "openssl_wrap.h"
+
+namespace mococrw
+{
+
+/**
+ * @brief The MessageAuthenticationCode class is the abstract base class for the different implementation of
+ * message authentication codes.
+ */
+class MessageAuthenticationCode {
+public:
+    /**
+     * Adds the message to the hash. This function may be invoked multiple times.
+     * For finishing the calculation of the MAC-tag invoke finish().
```

Comment:
```suggestion
     * For finishing the calculation of the MAC invoke finish().
```

I think "message authentication code" is the common term for both the algorithm and it's bytestring return value, so we should probably use MAC here. "tag" is only used in AES-GCM (and potentially other symmetric AEAD schemes) to my knowledge.